nohost/src/HTMLServer.js

```diff
@@ -0,0 +1,104 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define, brackets */
+define(function (require, exports, module) {
+    "use strict";
+
+    var BaseServer              = brackets.getModule("LiveDevelopment/Servers/BaseServer").BaseServer,
+        LiveDevelopmentUtils    = brackets.getModule("LiveDevelopment/LiveDevelopmentUtils"),
+        BlobUtils               = brackets.getModule("filesystem/impls/filer/BlobUtils"),
+        Filer                   = brackets.getModule("filesystem/impls/filer/BracketsFiler"),
+        Content                 = brackets.getModule("filesystem/impls/lib/content"),
+        Handlers                = brackets.getModule("filesystem/impls/lib/handlers"),
+        Rewriter                = brackets.getModule("filesystem/impls/lib/HTMLRewriter"),
+        Log                     = brackets.getModule("filesystem/impls/lib/log");
+
+    var Path = Filer.Path;
+
+    function HTMLServer(config) {
+        config = config || {};
+        BaseServer.call(this, config);
+    }
+
+    HTMLServer.prototype = Object.create(BaseServer.prototype);
+    HTMLServer.prototype.constructor = HTMLServer;
+
+    //Returns a pre-generated blob url based on path
+    HTMLServer.prototype.pathToUrl = function(path) {
+        return BlobUtils.getUrl(path);
+    };
+    //Returns a path based on blob url
+    HTMLServer.prototype.urlToPath = function(url) {
+        return BlobUtils.getFilename(url);
+    };
+
+    HTMLServer.prototype.start = function() {
+        this.fs = Filer.fs();
+    };
+
+    HTMLServer.prototype.stop = function() {
+        this.fs = null;
+    };
+
+    /**
+     * Determines if this server can serve local file. LiveDevServerManager
+     * calls this method when determining if a server can serve a file.
+     * @param {string} localPath A local path to file being served.
+     * @return {boolean} true When the file can be served, otherwise false.
+     */
+    HTMLServer.prototype.canServe = function (localPath) {
+        // If we can't transform the local path to a project relative path,
+        // the path cannot be served
+        if (localPath === this._pathResolver(localPath)) {
+            return false;
+        }
+
+        // Url ending in "/" implies default file, which is usually index.html.
+        // Return true to indicate that we can serve it.
+        if (localPath.match(/\/$/)) {
+            return true;
+        }
+
+        // FUTURE: do a MIME Type lookup on file extension
```

Comment:
Remove this comment, it doesn't apply to what we're doing here.  We always want HTML files.